core/commands/flag/interactors/delete_flag.py

```diff
@@ -21,8 +24,9 @@ def validate(self, owner: Owner, repo: Repository):
         if not repo:
             raise ValidationError("Repo not found")
 
-        if not owner.is_admin(self.current_user):
-            raise Unauthorized()
+        if settings.IS_ENTERPRISE:
+            if not self_hosted.is_admin_owner(self.current_user):
```

Comment:
I think this should take an `Owner` instance instead of a `User`

Reply:
`self.current_user` is an Owner object